tests/make_index_test.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+import io
+import json
+import os.path
+import urllib.request
+import zipfile
+from unittest import mock
+
+import re_assert
+
+import make_index
+
+
+def make_wheel(path, metadata):
+    name, v, *_ = os.path.basename(path).split("-")
+    with zipfile.ZipFile(path, "w") as zipf:
+        with zipf.open(f"{name}-{v}.dist-info/METADATA", "w") as f:
+            f.write(f"Name: {name}\n".encode())
+            f.write(f"Version: {v}\n".encode())
+            for k, v in metadata:
+                f.write(f"{k}: {v}\n".encode())
+
+
+def test_make_info_empty_wheel_metadata(tmp_path):
+    filename = str(tmp_path.joinpath("a-1-py3-none-any.whl"))
+    make_wheel(filename, ())
+
+    ret = make_index._make_info(filename)
+    assert ret == {
+        "filename": "a-1-py3-none-any.whl",
+        "hash": "sha256=64f7f4664408d711c17ad28c1d3ba7dd155501e67c8632fafc8a525ba3ebc527",
+        "upload_timestamp": mock.ANY,
+        "uploaded_by": re_assert.Matches(r"^git@[a-f0-9]{7}"),
+    }
+
+
+def test_make_info_full_wheel_metadata(tmp_path):
+    filename = str(tmp_path.joinpath("a-1-py3-none-any.whl"))
+    make_wheel(
+        filename,
+        (
+            ("Requires-Python", ">= 3.7, != 3.7.0"),
```

Comment:
3.8?

Reply:
I copied this from some random package -- can't remember which one now lol